Lessson16/ayjan/script.js

```diff
@@ -0,0 +1,667 @@
+   - Log "Starts with A" or "Does not start with A".
+*/
+function startsWithA(str) {
+  if (str.charAt(0) === "A") {
+    console.log("Starts with A");
+  } else {
+    console.log("Does not start with A");
+  }
+}
+startsWithA("Ayjan");
+startsWithA("ayjan");
+startsWithA("John");
+console.log("---------------------------------");
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n) {
+  const lastNCharacter = text.slice(-n);
+  console.log(lastNCharacter);
+}
+sliceLastN("Ayjan", 2);
+sliceLastN("JavaScript", 3);
+sliceLastN("My name is Ayjan", 8);
+console.log("---------------------------------");
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+function gradeChecker(score) {
+  if (score >= 90) {
+    console.log("A");
+  } else if (score >= 80) {
+    console.log("B");
+  } else if (score >= 70) {
+    console.log("C");
+  } else if (score >= 60) {
+    console.log("D");
+  } else {
+    console.log("F");
+  }
+}
+gradeChecker(30);
+gradeChecker(65);
+gradeChecker(85);
+console.log("---------------------------------");
+
+function gradeChecker(score) {
+  const sanitizedScore = parseInt(score);
+
+  if (typeof sanitizedScore !== "number" || isNaN(sanitizedScore)) {
+    console.log("Invalid input", score);
+    return;
+  }
+  switch (true) {
+    case sanitizedScore >= 90:
+      console.log("A");
+      break;
+    case sanitizedScore >= 80:
+      console.log("B");
+      break;
+    case sanitizedScore >= 70:
+      console.log("C");
+      break;
+    case sanitizedScore >= 60:
+      console.log("D");
+      break;
+    case sanitizedScore < 60:
+      console.log("F");
+      break;
+
+    default:
+      console.log("Provided number is outside the range", score);
+      break;
+  }
+}
+gradeChecker(30);
+gradeChecker(65);
+gradeChecker(85);
+console.log("---------------------------------");
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+function replaceCharacter(str, oldChar, newChar) {
+  const replacedStr = str.replaceAll(oldChar, newChar);
+  console.log(replacedStr);
+}
+replaceCharacter("flowers", "f", "F");
+replaceCharacter("hella", "a", "o");
+
+console.log("---------------------------------");
+
+/*19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word // burada map() methodu kullan.
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence) {
+  const words = sentence.split("");
```

Comment:
Should be a space inside the quotes to split the sentence into words
```suggestion
  const words = sentence.split(" ");
```